### ISSUE TITLE
ADI roll is opposite of what it should be

### ISSUE THREAD

**Issue by wiseman** [wiseman/mavelous#21]
Running mavelous and the Ardupilot Mission Planner in parallel today I noticed that the mavelous ADI shows the aircraft rolling in the opposite direction that the Mission Planner shows.

(Am I crazy, can anyone else confirm that it's wrong?)

**Comment by pgiacalo** [comment]
Hi John. I can confirm the wrong movement of the ADI for roll maneuvers first mentioned above. I grabbed a fresh copy of the code after the fix (05bdf91) and ran the same test but the problem was still there.  

Here's how I grabbed the code. Does this grab the latest version with the fixes? 

$ cd src
$ git clone git@github.com:mavlink/mavlink.git
$ git clone git@github.com:wiseman/mavelous.git

Also, I noticed that the command to launch mavelous on a Mac had to be changed, relative to the prior release. This was needed in order to get it to run. Should the documentation be updated to reflect this?

**Comment by wiseman** [comment]
Philip, that should get the latest code, including the fix, yes.  And you can confirm by checking [line 52 of modules/lib/mavelous_web/script/pfd.js](https://github.com/wiseman/mavelous/blob/master/modules/lib/mavelous_web/script/pfd.js#L52), which should look like this:

```
      context.rotate(-this.roll);
```

I'll test later today to check which behavior I see.

Can you open a new issue or send mail to mavelous@googlegroups about how to launch on OS X, including the old and new way?  I'm not positive I've ever tried it in OS X...

**Comment by pgiacalo** [comment]
Hi Johh,
Yes, I see the following line in the new code I'm running

      context.rotate(-this.roll);

**Comment by wiseman** [comment]
I just tested running mavelous and the ardupilot mission planner in parallel, connected to the same simulated arducopter, and saw the roll being displayed in mavelous consistent with the roll being show in the mission planner. As far as I can tell it's working correctly.

Maybe you can try running the two in parallel for confirmation?

**Comment by pgiacalo** [comment]
Great! Thanks for taking the time to perform that test. 
 
I'm running the UAV Dev Board with MatrixPilot 3.2 so cannot do the same exact test as you. But I will do some more tests and let you know the outcome.
 
Perhaps there's a bug in the implementation of the mavlink protocol for MatrixPilot. It's relatively new code, from what I understand. But before I draw that conclusion, I'd like to test it again. I'm wondering... can modify the line of code you mentioned before (line 52 pfd.js) and re-run mavelous before and after the code change. Will that work or does mavelous need to be recompiled to reflect the change? 
 
Thanks again and sorry for the false alarm.

**Comment by wiseman** [comment]
You can change the javascript code and then just hit reload in the browser to test it.  It's very convenient. :)

**Comment by pgiacalo** [comment]
Hi John,
The ADI in mavelous is working correctly. Thanks for that fix again. Not sure why I had the problem before but suspect the old code was still being invoked. 

One more question, if you don't mind. The black aircraft silhouette is always shown with the nose pointed towards the top of the screen. Is this a problem on my end or is aircraft heading/attitude a TBD feature? 

Thanks again for the great software!
Phil

**Comment by wiseman** [comment]
I think that's an issue on your end.  It's worked for me forever.  Are there any errors in your browser's javascript console?

**Comment by pgiacalo** [comment]
I tried it in Firefox, Chrome and Safari and the plane orientation is fixed and pointed up on all three. Here are the few messages I see in Firefox. 
 
Firefox Console Error:
Timestamp: 11/13/12 8:49:46 PM
Warning: Unknown property '-moz-user-drag'.  Declaration dropped.
Source File: http://127.0.0.1:9999/index.html#overview
Line: 0

Firefox Console Warning:
Timestamp: 11/13/12 8:49:46 PM
Warning: Unknown property '-moz-user-drag'.  Declaration dropped.
Source File: http://127.0.0.1:9999/index.html#overview
Line: 0

Any help?

**Comment by wiseman** [comment]
I don't think that warning should break things.  I typically use Chrome, for what it's worth.

The vehicle icon orientation comes from the mavlink ATTITUDE message.  You can see the current ATTITUDE by going to http://127.0.0.1:9999/mavlink/ATTITUDE in your browser.  It should look something like this:

```
{
  "ATTITUDE": {
    "msg": {
      "pitchspeed": -0.10704250633716583,
      "yaw": 0.10871666669845581,
      "rollspeed": 0.0165330208837986,
      "time_boot_ms": 54751,
      "pitch": -0.07376787811517715,
      "mavpackettype": "ATTITUDE",
      "yawspeed": -0.01481429673731327,
      "roll": 0.009263534098863602
    },
    "index": 34,
    "time_usec": 54711000
  }
}
```

I'm going to close this bug, but feel free to open another for the icon orientation, or post to mavelous@googlegroups.com.

**Comment by pgiacalo** [comment]
Ok. Thanks again, John. 

Just to correct something in one of my earlier posts. I'm running MatrixPilot v3.3 not 3.2
